utilities/copyoffload_migration.py

```
@@ -54,6 +57,123 @@ def get_copyoffload_credential(
+    """
+    extra: dict[str, str] = {}
+    if "storage_secret_extra" not in copyoffload_config:
+        extra.update(parse_storage_secret_extra_env())
+        return extra
+
+    config_extra = copyoffload_config["storage_secret_extra"]
+    if config_extra is None:
+        extra.update(parse_storage_secret_extra_env())
+        return extra
+    if not isinstance(config_extra, dict):
+        raise ValueError("storage_secret_extra must be a JSON object mapping Secret keys to values")
```

Ruff (0.15.14)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)